arch/arm/core/aarch32/cpu_idle.S

```diff
@@ -44,86 +44,119 @@ GTEXT(arch_cpu_atomic_idle)
 
 SECTION_FUNC(TEXT, z_arm_cpu_idle_init)
 #if defined(CONFIG_CPU_CORTEX_M)
-	ldr r1, =_SCB_SCR
-	movs.n r2, #_SCR_INIT_BITS
-	str r2, [r1]
+	ldr	r1, =_SCB_SCR
+	movs.n	r2, #_SCR_INIT_BITS
+	str	r2, [r1]
 #endif
-	bx lr
+	bx	lr
 
 SECTION_FUNC(TEXT, arch_cpu_idle)
 #ifdef CONFIG_TRACING
-	push {r0, lr}
-	bl    sys_trace_idle
+	push	{r0, lr}
+	bl	sys_trace_idle
 #if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
-	pop {r0, r1}
-        mov lr, r1
+	pop	{r0, r1}
+	mov	lr, r1
 #else
-	pop {r0, lr}
+	pop	{r0, lr}
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 #endif /* CONFIG_TRACING */
 
-#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE) \
-	|| defined(CONFIG_ARMV7_R)
-	cpsie i
-#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
-	/* clear BASEPRI so wfi is awakened by incoming interrupts */
-	eors.n r0, r0
-	msr BASEPRI, r0
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+	/*
+	 * PRIMASK is always cleared on ARMv7-M and ARMv8-M Mainline, and
```

Comment:
Maybe add the reasoning here, too, as you did in line 86.
E.g "PRIMASK is ....... Mainline (not used for interrupt locking), and ..."

Reply:
Done